Managers/UInputMan.h

```diff
@@ -620,33 +651,40 @@ namespace RTE {
 		void ClearNetworkAccumulatedStates();
 #pragma endregion
 
-	protected:
+	private:
 
 		/// <summary>
 		/// Enumeration for the different states an input element or button can be in.
 		/// </summary>
 		enum InputState { Held, Pressed, Released, InputStateCount };
 
-		static GUIInput *s_GUIInputInstanceToCaptureKeyStateFrom; //!< Pointer to the GUIInput instance to capture key state from, if any. This is used for better key detection during input mapping input capture.
+		static std::array<uint8_t, SDL_NUM_SCANCODES> s_PrevKeyStates; //!< Key states as they were the previous update.
```

Comment:
What's the general benefit of using scancodes instead of keycodes in places? Seems to me like scancodes are pretty useless for the end user vs keycodes, and both seem to be usable internally?
I'm curious

Reply:
In this specific case it's having an array of 512 bytes vs int_max bytes.
keycodes map according to the keyboard layout (qerty, azerty, dvorak)  and scancodes map physical locations (keymap independent), so it's to have certain things map to physical locations:
- Console key
- All presets (makes wasd, etc. map to the right keys on cursed keymaps like french)

Reply:
Okay, that makes sense. Those are very reasonable situations where physical key location matters rather than the actual key.
Thanks!